### PR TITLE
Fix various problems in the pbench_results_push tests

### LINKED ISSUE #3374: Testing fixes for test_results_push.py
There are some existing issues and some  arising from PR #3348 that should be addressed in the `pbench-results-push` tests (see [test_resuls_push.py](https://github.com/distributed-system-analysis/pbench/blob/main/lib/pbench/test/unit/agent/task/test_results_push.py)):

- Some tests (e.g. the tests that test exceptions) can be subsumed under the parametrization of `add_http_mock_response` with suitable changes so that the latter can accept an exception as well as a string for the `message`). That will eliminate `add_connectionerr_mock_response` and the individual tests that use it.

- [It might be better](https://github.com/distributed-system-analysis/pbench/pull/3348#discussion_r1161823694) to specify the numeric value (400) rather than a symbolic name that happens to have the same value when deciding how to format the error message.

- See also [this suggestion](https://github.com/distributed-system-analysis/pbench/pull/3348#discussion_r1161811397)  for modifying the parametrization of the tests.

### PR DESCRIPTION
**Pull request by ndokos**
Fixes #3374

This is a continuation of #3348, implementing fixes to the pbench_results_push tests. The main one is to subsume some exception tests under the parametrized "normal" case and eliminate redundant tests.

### CODE REVIEW

**Review by dbutenhof — DISMISSED** [review]
Nice cleanup; just a few consistency suggestions...

File: lib/pbench/test/unit/agent/task/test_results_push.py [hunk 1]
```diff
@@ -162,46 +149,105 @@ def test_multiple_meta_args_single_option():
...
-            (HTTPStatus.CREATED, None, 0),
-            (HTTPStatus.OK, {"message": "Dup"}, 0),
-            (HTTPStatus.OK, "Dup", 0),
-            (HTTPStatus.NO_CONTENT, {"message": "No content"}, 0),
-            (HTTPStatus.NO_CONTENT, "No content", 0),
+            (
+                HTTPStatus.CREATED,
+                None,
+                0,
+            ),
```
Comment (webbnh) [inline lib/pbench/test/unit/agent/task/test_results_push.py:201]: Did you add the trailing comma on line 200 (and 205, et al.)?  When I remove them, Black formats all but two of the tuples on single lines, which I think is easier to read (it re-added the ones for the wrapped lines, so perhaps you provoked it in some way?).

File: lib/pbench/test/unit/agent/task/test_results_push.py [hunk 2]
```diff
@@ -162,46 +149,105 @@ def test_multiple_meta_args_single_option():
...
+                "Not Found",
+                1,
+            ),
+            (
+                0,
+                requests.exceptions.ConnectionError(
+                    "<urllib3.connection.HTTPConnection object at 0x1080854c0>: "
+                    "Failed to establish a new connection: [Errno 8] "
+                    "nodename nor servname provided, or not known"
+                ),
```
Comment (webbnh) [inline lib/pbench/test/unit/agent/task/test_results_push.py:248]: I realize that this is basically a lift-and-shift, but unit tests don't _have_ to reflect reality...so I suspect that replacing this with
```python
                requests.exceptions.ConnectionError("Ooops")
```
would work fine.

File: lib/pbench/test/unit/agent/task/test_results_push.py [hunk 3]
```diff
@@ -221,118 +267,21 @@ def test_push_status(status_code, message, exit_code):
...
-            HTTPStatus.NOT_FOUND, message={"message": "Not Found"}
-        )
-        caplog.set_level(logging.DEBUG)
-        runner = CliRunner(mix_stderr=False)
-        result = runner.invoke(main, args=[tarball])
-        assert result.exit_code == 1
-        assert (
-            "Not Found" in result.stderr
-        ), f"stderr: {result.stderr!r}; stdout: {result.stdout!r}"
+        assert err_msg in result.stderr.strip()
```
Comment (webbnh) [inline lib/pbench/test/unit/agent/task/test_results_push.py:287]: Given that we're using `in`, I don't think we need the `.strip()`.

**Review by webbnh — APPROVED** [review]
I concur with Dave's assessment -- nice clean-up -- and his comments; and here are a few more nits, if you're so inclined.

File: lib/pbench/test/unit/agent/task/test_results_push.py [hunk 4]
```diff
@@ -162,46 +149,105 @@ def test_multiple_meta_args_single_option():
...
+                {"message": "Not Found"},
+                1,
+            ),
+            (
+                HTTPStatus.NOT_FOUND,
+                "Not Found",
+                1,
+            ),
+            (
+                0,
```
[the remark this answers is not in the document]
Comment (ndokos) [reply lib/pbench/test/unit/agent/task/test_results_push.py:243]: I started with `None` but somebody complained - let's see: it was the comparison with 400 that complained, so I just check for `None` explicitly now.

File: lib/pbench/test/unit/agent/task/test_results_push.py (hunk 1 above)
Comment (ndokos) [reply lib/pbench/test/unit/agent/task/test_results_push.py:201]: It's easy to provoke `black` to anger, so it is no fun: I had added an extra field and then took the extra field away but not the comma. I'm agnostic as to what is easier to read: I'd rather have one line and the fields aligned (but black does not like that either). Anyway, with the shortening of the exception message, only one entry is multiline.

**Review by webbnh — APPROVED** [review]
👍

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**